rpc/src/responses/bulk_array.rs

```diff
@@ -0,0 +1,31 @@
+use serde_json::{self, json};
+
+pub struct BulkArray<T> {
+    flattened_vec: Vec<T>,
+}
+
+impl BulkArray<String> {
+    pub fn from_str(str_to_parse: &str) -> Self {
+        let parse_response = serde_json::from_str(str_to_parse).unwrap_or_else(|_| json!([]));
+        let parsed_bulk_array = parse_response.as_array().unwrap();
+
+        let mut flattened_vec = Vec::new();
+        for nested_entry in parsed_bulk_array {
+            let unwrapped_item = nested_entry
```

Comment:
It might be safer to do `if nested_entry.is_array()` before `.as_array()` incase the `str_to_parse` is of form `["blockId1"]`, if it is not an array then we can just do `nested_entry.as_str().unwrap().to_string();`

Reply:
Like some bulk arrays, not all 1d arrays contain strings so this might introduce more complexity so please ignore this comment